File: network-configuration-server/src/main/java/en/ubb/networkconfiguration/boundary/dto/network/setup/NetworkInitDto.java
package en.ubb.networkconfiguration.boundary.dto.network.setup;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
public class NetworkInitDto {

    @JsonProperty("id")
    private long id;

    @JsonProperty("name")
    private String name;

    @JsonProperty("seed")
    private int seed;

    @JsonProperty("learningRate")
    private double learningRate;

    @JsonProperty("batchSize")
    private int batchSize;

    @JsonProperty("nEpochs")
    private int nEpochs;

    @JsonProperty("nInputs")
    private int nInputs;

    @JsonProperty("nOutputs")
    private int nOutputs;

    @JsonProperty
    private Long branchId;

    @JsonProperty("layers")
    private List<LayerInitDto> layers;
}

File: client-app/src/app/shared/models/network/setup/NetworkInitDto.ts
import {LayerInitDto} from './LayerInitDto';

export interface NetworkInitDto {
  id: number;
  name: string;
  seed: number;
  learningRate: number;
  batchSize: number;
  nEpochs: number;
  nInputs: number;
  nOutputs: number;
  branchId: number;
  layers: LayerInitDto[];
}
